### PR TITLE
Use conftest instead of opa for the EC

### PR DESCRIPTION
**Pull request by simonbaird**
Based on #105.

### CODE REVIEW

File: appstudio-utils/util-scripts/lib/fetch/cluster.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ save-policy-config() {
     fi
   else
     local non_blocking_data_file
-    non_blocking_data_file="$( json-data-file config policy non_blocking_checks)"
+    non_blocking_data_file=="$DATA_DIR/config.json"
```
Comment (simonbaird) [inline appstudio-utils/util-scripts/lib/fetch/cluster.sh:58]: Oops, will fix this ==

Comment (joejstuart) [reply appstudio-utils/util-scripts/lib/fetch/cluster.sh:58]: Is there even a reason for the variable? Below could just run
```
echo "$non_blocking_data" | jq > "$DATA_DIR/config.json"
```

Comment (joejstuart) [reply appstudio-utils/util-scripts/lib/fetch/cluster.sh]: nvm. I see it's used later on in the function also.

**Review by joejstuart — APPROVED** [review]
Looks good!

**Review by joejstuart — APPROVED** [review]
Looks good!

**Comment by simonbaird** [comment]
Here's what I'm doing to exercise this in ./hack:

```
env MY_QUAY_USER=$USER ./build-and-push.sh && ./start-verify-ec-task.sh quay.io/sbaird/single-nodejs-app:4921909 nodejs-builder-run-5t5fp
```

You'll need to have your own image and pipelinerun prepared.

File: appstudio-utils/Dockerfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ RUN curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/lat
 RUN curl -L https://github.com/sigstore/cosign/releases/download/v1.8.0/cosign-linux-amd64 -o /usr/bin/cosign && chmod +x /usr/bin/cosign
 RUN curl -L https://mirror.openshift.com/pub/openshift-v4/clients/pipeline/0.21.0/tkn-linux-amd64-0.21.0.tar.gz | tar -xz --no-same-owner -C /usr/bin/
 RUN curl -L https://github.com/sigstore/rekor/releases/download/v0.5.0/rekor-cli-linux-amd64 -o /usr/bin/rekor-cli && chmod +x /usr/bin/rekor-cli
+RUN curl -L https://github.com/open-policy-agent/conftest/releases/download/v0.32.0/conftest_0.32.0_Linux_x86_64.tar.gz | tar -xz --no-same-owner -C /usr/bin
 
 RUN curl -L https://github.com/open-policy-agent/opa/releases/download/v0.40.0/opa_linux_amd64_static -o /usr/bin/opa && \
     echo 73e96d8071c6d71b4a9878d7f55bcb889173c40c91bbe599f9b7b06d3a472c5f /usr/bin/opa | sha256sum --check && \
```
Comment (zregvart) [inline appstudio-utils/Dockerfile:12]: Should we now remove `opa`?

File: appstudio-utils/util-scripts/lib/fetch/cluster.sh [hunk 3]
```diff
@@ -54,13 +59,14 @@ save-policy-config() {
       return 1
     fi
   else
-    local non_blocking_data_file
-    non_blocking_data_file="$( json-data-file config policy non_blocking_checks)"
-    echo "$non_blocking_data" | jq > "${non_blocking_data_file}"
+    # Save the config data from the ECP
+    echo "$config_data" | jq '{"config": {"policy": . }}' > "${config_file}"
```
Comment (zregvart) [inline appstudio-utils/util-scripts/lib/fetch/cluster.sh:63]: We're only fetching the non-blocking configuration from EnterpriseContractPolicies custom resource above, so it needs to be nested under `config.policy.nonBlocking`:

```suggestion
    echo "$config_data" | jq '{"config": {"policy": {"non_blocking_checks": . }}}' > "${config_file}"
```

File: appstudio-utils/util-scripts/lib/fetch/data.sh [hunk 4]
```diff
@@ -16,14 +16,49 @@ json-data-file() {
...
   echo "$file"
 }
 
+# Emulate in-place editing with jq
+jq-in-place-edit() {
+  local jq_filter="$1"
+  local file="$2"
+  local tmp_file="$(mktemp)"
+
+  jq "$jq_filter" "$file" > "$tmp_file"
```
Comment (zregvart) [inline appstudio-utils/util-scripts/lib/fetch/data.sh:33]: Not sure we're handling errors here (similar to #84), to be on the safe side I'd do:
```suggestion
  jq "$jq_filter" "$file" > "$tmp_file" || return 1
```

Comment (simonbaird) [reply appstudio-utils/util-scripts/lib/fetch/data.sh:33]: I went for this:

```
 33   jq "$jq_filter" "$file" > "$tmp_file" && \                                                                                                                                                                                             
 34     mv "$tmp_file" "$file" 
```

File: appstudio-utils/util-scripts/lib/fetch/cluster.sh (hunk 3 above)
Comment (simonbaird) [reply appstudio-utils/util-scripts/lib/fetch/cluster.sh:63]: I might have found that if my cluster actually had the ECP resource working.

Comment (simonbaird) [reply appstudio-utils/util-scripts/lib/fetch/cluster.sh:63]: I realized I changed the var name based on a misunderstanding. Will change it back in next revision.

**Review by robnester-rh — APPROVED** [review]
LGTM

**Pull request merged**